### PR TITLE
feat: add feature flag storage based on Redis and maintenance mode based on feature flag

### PR DESCRIPTION
**Pull request by sumandas0**
## Feature flag storage powered by Redis

> Add an Redis based feature flag to apply feature flag without restarting the pod, this will will help us in multiple occasion where we need do this, like to block write functionality but without any delay of the pod restart.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by sumandas0** [comment]
Add feature flag `maintenance_mode = true`
```
curl --location --request POST 'http://localhost:21000/api/atlas/v2/admin/featureFlag?key=maintenance_mode&value=true' 
```
Delete feature flag, `maintenance_mode`
```
curl --location --request DELETE 'http://localhost:21000/api/atlas/v2/admin/featureFlag/maintenance_mode' 
```

File: intg/src/main/java/org/apache/atlas/AtlasConfiguration.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ public enum AtlasConfiguration {
 
     INDEXSEARCH_ASYNC_SEARCH_KEEP_ALIVE_TIME_IN_SECONDS("atlas.indexsearch.async.search.keep.alive.time.in.seconds", 300),
 
-    ATLAS_MAINTENANCE_MODE("atlas.maintenance.mode", false);
+    ATLAS_MAINTENANCE_MODE("atlas.maintenance.mode", true);
```
Comment (mehtaanshul) [inline intg/src/main/java/org/apache/atlas/AtlasConfiguration.java:116]: This should stay "false", right?

Comment (sumandas0) [reply intg/src/main/java/org/apache/atlas/AtlasConfiguration.java]: Yes

File: repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java [hunk 2]
```diff
@@ -1151,7 +1152,7 @@ private String getIndexName(IndexSearchParams params) throws AtlasBaseException
         if (StringUtils.isNotEmpty(aliasName)) {
             if(params.isAccessControlExclusive()) {
                 accessControlExclusiveDsl(params, aliasName);
-                aliasName = aliasName+","+VERTEX_INDEX_NAME;
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java:1154]: Make sure you have test case to test this flow.

File: common/src/main/java/org/apache/atlas/repository/Constants.java [hunk 3]
```diff
@@ -453,6 +455,19 @@ private static String getEncodedTypePropertyKey(String defaultKey) {
...
     }
 
+    public static String getESIndex() {
+        String indexSuffix  = null;
+        if(AtlasConfiguration.ATLAS_MAINTENANCE_MODE.getBoolean()) {
+            try {
+                if (FeatureFlagStore.evaluate("use_temp_es_index", "true")) {
+                    indexSuffix = "_temp";
+                }
+            } catch (Exception ignored) {
```
Comment (mehtaanshul) [inline common/src/main/java/org/apache/atlas/repository/Constants.java:465]: Add a log in "catch" here.

File: repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java (hunk 2 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java:1154]: Yes

File: common/src/main/java/org/apache/atlas/repository/Constants.java (hunk 3 above)
Comment (sumandas0) [reply common/src/main/java/org/apache/atlas/repository/Constants.java]: Done

**Review by mehtaanshul — APPROVED** [review]

**Pull request merged**